packages/Search/test/tstDetailsAlgorithms.cpp

```diff
@@ -31,58 +31,81 @@ TEUCHOS_UNIT_TEST( DetailsAlgorithms, distance )
-    TEST_ASSERT( !dtk::overlaps( box, DataTransferKit::Box() ) );
+    TEST_ASSERT( dtk::intersects( box, box ) );
+    TEST_ASSERT( !dtk::intersects( box, DataTransferKit::Box() ) );
     // overlap with box that contains it
     TEST_ASSERT(
-        dtk::overlaps( box, {{{-1.0, -1.0, -1.0}}, {{1.0, 1.0, 1.0}}} ) );
+        dtk::intersects( box, {{{-1.0, -1.0, -1.0}}, {{1.0, 1.0, 1.0}}} ) );
     // does not overlap with some other box
     TEST_ASSERT(
-        !dtk::overlaps( box, {{{1.0, 1.0, 1.0}}, {{2.0, 2.0, 2.0}}} ) );
+        !dtk::intersects( box, {{{1.0, 1.0, 1.0}}, {{2.0, 2.0, 2.0}}} ) );
     // overlap when only touches another
-    TEST_ASSERT( dtk::overlaps( box, {{{0.0, 0.0, 0.0}}, {{1.0, 1.0, 1.0}}} ) );
+    TEST_ASSERT(
+        dtk::intersects( box, {{{0.0, 0.0, 0.0}}, {{1.0, 1.0, 1.0}}} ) );
     // unit cube
     box = {{{0.0, 0.0, 0.0}}, {{1.0, 1.0, 1.0}}};
-    TEST_ASSERT( dtk::overlaps( box, box ) );
-    TEST_ASSERT( !dtk::overlaps( box, DataTransferKit::Box() ) );
+    TEST_ASSERT( dtk::intersects( box, box ) );
+    TEST_ASSERT( !dtk::intersects( box, DataTransferKit::Box() ) );
     // smaller box inside
     TEST_ASSERT(
-        dtk::overlaps( box, {{{0.25, 0.25, 0.25}}, {{0.75, 0.75, 0.75}}} ) );
+        dtk::intersects( box, {{{0.25, 0.25, 0.25}}, {{0.75, 0.75, 0.75}}} ) );
     // bigger box that contains it
     TEST_ASSERT(
-        dtk::overlaps( box, {{{-1.0, -1.0, -1.0}}, {{2.0, 2.0, 2.0}}} ) );
+        dtk::intersects( box, {{{-1.0, -1.0, -1.0}}, {{2.0, 2.0, 2.0}}} ) );
     // couple boxes that do overlap
-    TEST_ASSERT( dtk::overlaps( box, {{{0.5, 0.5, 0.5}}, {{1.5, 1.5, 1.5}}} ) );
     TEST_ASSERT(
-        dtk::overlaps( box, {{{-0.5, -0.5, -0.5}}, {{0.5, 0.5, 0.5}}} ) );
+        dtk::intersects( box, {{{0.5, 0.5, 0.5}}, {{1.5, 1.5, 1.5}}} ) );
+    TEST_ASSERT(
+        dtk::intersects( box, {{{-0.5, -0.5, -0.5}}, {{0.5, 0.5, 0.5}}} ) );
     // couple boxes that do not
     TEST_ASSERT(
-        !dtk::overlaps( box, {{{-2.0, -2.0, -2.0}}, {{-1.0, -1.0, -1.0}}} ) );
+        !dtk::intersects( box, {{{-2.0, -2.0, -2.0}}, {{-1.0, -1.0, -1.0}}} ) );
     TEST_ASSERT(
-        !dtk::overlaps( box, {{{0.0, 0.0, 2.0}}, {{1.0, 1.0, 3.0}}} ) );
+        !dtk::intersects( box, {{{0.0, 0.0, 2.0}}, {{1.0, 1.0, 3.0}}} ) );
     // boxes overlap if faces touch
-    TEST_ASSERT( dtk::overlaps( box, {{{1.0, 0.0, 0.0}}, {{2.0, 1.0, 1.0}}} ) );
     TEST_ASSERT(
-        dtk::overlaps( box, {{{-0.5, -0.5, -0.5}}, {{0.5, 0.0, 0.5}}} ) );
+        dtk::intersects( box, {{{1.0, 0.0, 0.0}}, {{2.0, 1.0, 1.0}}} ) );
+    TEST_ASSERT(
+        dtk::intersects( box, {{{-0.5, -0.5, -0.5}}, {{0.5, 0.0, 0.5}}} ) );
+}
+
+TEUCHOS_UNIT_TEST( DetailsAlgorithms, intersects )
+{
+    DataTransferKit::Sphere sphere = {{{0., 0., 0.}}, 1.};
+    TEST_ASSERT( dtk::intersects( sphere, {{{0., 0., 0.}}, {{1., 1., 1.}}} ) );
+    TEST_ASSERT( !dtk::intersects( sphere, {{{1., 2., 3.}}, {{4., 5., 6.}}} ) );
```

Comment:
Do we have a sphere-sphere intersection or only sphere-box because we are checking intersection with tree nodes?

Reply:
This is sphrere-box intersection.  This is the only one needed to determine whether a sphere intersects with the nodes of the tree.  I did not bother writing sphere-sphere since it is not needed at this point.